Public/Invoke-Locksmith.ps1

```diff
@@ -38,26 +38,61 @@
     Finds any malconfigurations and creates code snippets to fix each issue.
     Attempts to fix all identified issues. This mode may require high-privileged access.
 
-    .INPUTS
-    None. You cannot pipe objects to Invoke-Locksmith.ps1.
+    .PARAMETER Action
+    Specify the desired action to perform:
+
+     - Scan: (Default) Find common ADCS malconfigurations and report the results.
+     - Fix: Scan ADCS for common malconfigurations and then prompt interactively to fix each individual finding.
+
+    .PARAMETER Output
+    Specify the desired output content:
+
+     - Findings: (Default) List all malconfigurations found in ADCS.
+     - DetailedFindings: List all findings with a detailed explanation of each type, including the risk and the resolution.
+     - RemediationScript: Create a script that will help remediate all findings.
+    
+    .PARAMETER OutputType
+    Specify the type of output or file you want to create after scanning. Allows use of multiple values to output multiple file types.
+
+     - Host     (Default) Output the details to your current host console.
+     - CSV      Create a CSV file with the details of found malconfigurations.
+     - HTML     Create an HTML report.
+     - PDF      Create a PDF report.
+
+    .PARAMETER OutputPath
+    Specify the path where you want to save reports and mitigation scripts.
 
     .OUTPUTS
     Output types:
     1. Console display of identified issues
     2. Console display of identified issues and their fixes
     3. CSV containing all identified issues
     4. CSV containing all identified issues and their fixes
-    #>
 
-    # Windows PowerShell cmdlet Restart-Service requires RunAsAdministrator
+    .INPUTS
+    None. You cannot pipe objects to Invoke-Locksmith.
+
+    .NOTES
+    Windows PowerShell cmdlet Restart-Service requires RunAsAdministrator
+    #>
 
     [CmdletBinding()]
     param (
         [string]$Forest,
         [string]$InputPath,
-        [int]$Mode = 0,
-        [string]$OutputPath = (Get-Location).Path,
-        [System.Management.Automation.PSCredential]$Credential
+        [string]$Mode = 0,
+        [System.Management.Automation.PSCredential]$Credential,
+        [Parameter()]
+            [ValidateSet('Auditing','ESC1','ESC2','ESC3','ESC4','ESC5','ESC6','ESC8','All','PromptMe')]
+            [array]$Scans = 'All',
+        [Parameter()]
+            [ValidateSet('Findings','Detailed', ErrorMessage = 'Please choose Findings or Detailed for your output. Detailed output includes the fixes.')]
```

Comment:
`ErrorMessage` is only valid in Powershell 6.2ish and higher (https://stackoverflow.com/questions/72103714/unable-to-utilize-errormessage-property-with-validateset-in-function). Since we are targeting Windows Powershell AND Powershell Core, you can replace `ValidateSet` with `ValidateScript` OR remove `ErrorMessage`. Your call!

Reply:
Hmmm, you're right. Another great catch. (I tried.) 🥁

Reply:
ValidateSet is already present, so I'll just remove ErrorMessage for now.